sklearn/ensemble/_gradient_boosting.pyx

```diff
@@ -44,6 +46,7 @@ cdef void _predict_regression_tree_inplace_fast(DTYPE_t *X,
                                                 Py_ssize_t K,
                                                 Py_ssize_t n_samples,
                                                 Py_ssize_t n_features,
+                                                INT32_t* n_categories,
```

Comment:
Well, I'm using -1 to stand for non-categorical features, so it needs to be signed. Otherwise, no reason.